File: java/shared/java/org/signal/libsignal/protocol/kem/KEMKeyType.java
//
// Copyright 2023 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//
package org.signal.libsignal.protocol.kem;

public enum KEMKeyType {
  // Make sure to update KEMKeyPair.generate when adding new key types
  KYBER_1024(8);

  private final int type;

  private KEMKeyType(int type) {
    this.type = type;
  }
}

File: java/shared/java/org/signal/libsignal/protocol/kem/KEMKeyType.kt
//
// Copyright 2023 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

package org.signal.libsignal.protocol.kem

public enum class KEMKeyType(
  public val type: Int,
) {
  KYBER_1024(8),
}
